torch_geometric/distributed/dist_loader.py

```diff
@@ -0,0 +1,103 @@
+import atexit
+import os
+from typing import Dict, List, Optional, Union
+
+import torch
+import torch.multiprocessing as mp
+
+from .dist_context import DistContext, DistRole
+from .dist_neighbor_sampler import DistNeighborSampler, close_sampler
+from .rpc import global_barrier, init_rpc
+
+
+class DistLoader():
+    def __init__(
+            self,
+            neighbor_sampler: DistNeighborSampler,
```

Comment:
Why do we desperately need the `neighbor_sampler` argument here? I would imagine that a `DistLoader` class is much more general.

Reply:
fixed